Urban-Fun/View Controllers/SearchViewController.m

```diff
@@ -6,26 +6,225 @@
+@property (strong, nonatomic) IBOutlet UIBarButtonItem *filtersButton;
+@property (strong, nonatomic) IBOutlet UISegmentedControl *searchControl;
+@property (nonatomic, strong) NSMutableArray<User *> *usersArray;
+@property (nonatomic, strong) NSMutableArray<Activity *> *activitiesArray;
+@property (nonatomic, strong) User *profileToView;
+@property (nonatomic, strong) const CLLocationManager *manager;
+@property (nonatomic, strong) CLLocation *currentUserLocation;
 @end
 
 @implementation SearchViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.searchBar.delegate = self;
+    
+    // dismisses keyboard when tap outside a text field
+    UITapGestureRecognizer *tapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self.view action:@selector(endEditing:)];
+    [tapGestureRecognizer setCancelsTouchesInView:NO];
+    [self.view addGestureRecognizer:tapGestureRecognizer];
+    // dismisses keyboard when user scrolls
+    self.tableView.keyboardDismissMode = UIScrollViewKeyboardDismissModeOnDrag;
+    
     // Do any additional setup after loading the view.
+    UIFont *font = [UIFont fontWithName:@"Verdana-Bold" size:12.0f];
+    UIColor *color = [UIColor lightGrayColor];
+    NSMutableDictionary *attributes = [NSMutableDictionary dictionaryWithObject:color forKey:NSForegroundColorAttributeName];
+    [attributes setObject:font forKey:NSFontAttributeName];
+    
+    [self.searchControl setTitleTextAttributes:attributes
+                                    forState:UIControlStateSelected];
+    UIColor *color2 = [UIColor colorWithRed:68.0/255.0 green:85.0/255.0 blue:96.0/255.0 alpha:1.0];
+    NSMutableDictionary *attributesForNormal = [NSMutableDictionary dictionaryWithObject:color2 forKey:NSForegroundColorAttributeName];
+    [attributesForNormal setObject:font forKey:NSFontAttributeName];
+    [self.searchControl setTitleTextAttributes:attributesForNormal
+                                    forState:UIControlStateNormal];
+    self.manager = [[CLLocationManager alloc] init];
+    self.manager.delegate = self;
+    self.manager.desiredAccuracy = kCLLocationAccuracyHundredMeters; //battery
+    [self.manager requestWhenInUseAuthorization];
+    if (CLLocationManager.locationServicesEnabled){
+        [self.manager startUpdatingLocation];
+    }
+}
+
+- (void) locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray<CLLocation *> *)locations{
+    if (locations.firstObject){
+        [manager stopUpdatingLocation];
+        self.currentUserLocation = locations.firstObject;
+    }
+}
+
+- (IBAction)changedSearchType:(id)sender {
+    NSString *searchText = self.searchBar.text;
+    if (self.searchControl.selectedSegmentIndex == 0){
+        //removing white space
+        searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
```

Comment:
For a next iteration: do you think we can make this more abstract? Think about a helper class for the API where you can pass down the parameters you want to query and then the return value will be the array of activities or users, since you will be adding more logic for the filters I think is better to have it.

One step further is to use generics to pass down the type of object you want to query and the return will be the same type of object, you can take a look [here](https://medium.com/ios-os-x-development/generics-in-objective-c-8f54c9cfbce7)